EventSourcing.Cosmos/StoredProcedures/DeleteAggregateAll.js

```diff
@@ -0,0 +1,117 @@
+    function getIndex() {
+        var versionQuery = `SELECT Max(e.Index) AS Index FROM ${containerId} e WHERE e.PartitionId = '${partitionId}' AND e.AggregateId = '${aggregateId}'`;
+        var isAccepted = collection.queryDocuments(collectionLink, versionQuery, {}, createReservation);
+    }
+
+    // Create reservation event to prevent concurrency issues when deleting
+    function createReservation(err, retrievedDocs, responseOptions) {
+        if (err) throw err;
+        var index = retrievedDocs[0].Index >= 0 ? retrievedDocs[0].Index + 1 : 0;
+        var isAccepted = collection.createDocument(
+            collectionLink,
+            {
+                id: `Event|${aggregateId}[${index}]`,
+                PartitionId: partitionId,
+                AggregateId: aggregateId,
+                Index: index},
+            {},
+            deleteReservation);
+    }
+
+    // Delete reservation
+    function deleteReservation(err, resource, responseOptions) {
+        if (err) throw err;
+        var isAccepted = collection.deleteDocument(resource._self, {}, (err, resource, responseOptions) => {
+            if (err) throw err;
+            // Continue with querying and deleting
+            tryQueryAndDelete();
+        });
+    }
+    
+    // Recursively runs the query w/ support for continuation tokens.
+    // Calls tryDelete(documents) as soon as the query returns documents.
+    function tryQueryAndDelete(continuation) {
+        var requestOptions = {continuation: continuation};
+
+        var isAccepted = collection.queryDocuments(collectionLink, query, requestOptions,  (err, retrievedDocs, responseOptions) => {
+            if (err) throw err;
+
+            if (retrievedDocs.length > 0) {
+                // Begin deleting documents as soon as documents are returned form the query results.
+                // tryDelete() resumes querying after deleting; no need to page through continuation tokens.
+                //  - this is to prioritize writes over reads given timeout constraints.
+                tryDelete(retrievedDocs);
+            } else if (responseOptions.continuation) {
+                // Else if the query came back empty, but with a continuation token; repeat the query w/ the token.
+                tryQueryAndDelete(responseOptions.continuation);
+            } else {
+                // Else if there are no more documents and no continuation token - we are finished deleting documents.
+                responseBody.continuation = false;
+                response.setBody(responseBody);
+            }
+        });
+
+        // If we hit execution bounds - return continuation: true.
+        if (!isAccepted) {
+            response.setBody(responseBody);
+        }
+    }
+
+    // Recursively deletes documents passed in as an array argument.
+    // Attempts to query for more on empty array.
+    function tryDelete(documents) {
```

Comment:
Same as with tryQueryAndDelete, how deep does this recurse and should this be replaced with iteration?